ui/app/(ai)/cache/lib/cache.ts

```diff
@@ -0,0 +1,219 @@
+"use server";
+
+import { getProviders } from "@/actions/providers/providers";
+import { getScans } from "@/actions/scans/scans";
+import { getProfileInfo } from "@/actions/users/users";
+
+// Cache storage
+type CacheStore = {
+  [userId: string]: {
+    data: CachedData;
+    timestamp: number;
+  };
+};
+
+// In-memory cache store
+const cacheStore: CacheStore = {};
+
+// We'll use this to track the cache metadata
+let cacheVersion = Date.now();
+let cacheCreatedAt = new Date().toISOString();
+let cacheHits = 0;
+let cacheMisses = 0;
+
+// Type definition for our cached data
+interface CachedData {
+  user: {
+    name: string;
+    email: string;
+    company: string;
+  };
+  providers: Array<{
+    name: string;
+    provider_type: string;
+    alias: string;
+    id: string;
+    last_checked_at: string;
+    scan_id?: string;
+    scan_duration?: string;
+    resource_count?: number;
+  }>;
+}
+
+// Function to fetch all required data from APIs
+const fetchDataFromAPIs = async (): Promise<CachedData> => {
```

Comment:
Following up on this comment: https://github.com/prowler-cloud/prowler/pull/7508#discussion_r2052086824 

Could we please use the `/app` folder only for routing purposes?

Thank you!

Reply:
Moved this code to `/lib/lighthouse/cache.ts`.